clients/source.go

```diff
@@ -195,6 +197,26 @@ func (c *SourceClient) GetTables(ctx context.Context) ([]*schema.Table, error) {
 	return tables, nil
 }
 
+func (c *SourceClient) Validate(ctx context.Context, spec specs.Source) (warnings, errors []string, err error) {
```

Comment:
Apart from the errors shadow/nit, could maybe consider returning a ValidateResult struct instead of the triple return values? But then it would be very similar to `ValidateDestination_Response`. Maybe there's a better way? Pass a logger (as interface) and let it log? It's a bunch of lines, wouldn't need any formatting on the CLI side anyway.

Reply:
Yeah I'd be good with a struct; it's more extensible in the long run too. I don't think it's an issue that it's similar to `ValidateDestination_Response`.